### PR TITLE
Google Analytics Integated

### PR DESCRIPTION
**Pull request by ArpitKotecha**
Change google tag id (UA-136271091-1) with the id from the account you want the insights on or comment your email so I can add your account to the existing tag.

### CODE REVIEW

**Comment by vinitshahdeo** [comment]
That's awesome @ArpitKotecha !
Once I'll get the credentials from @ShreyaAnand, I'll update `Google Analytics ID` and merge this.

Can you update the same for [vinnovateit.com](https://vinnovateit.com/) too!

**Review by vinitshahdeo — CHANGES_REQUESTED** [review]
@ArpitKotecha, **add the script at end!**

Running `javascript` first may potentially cause performance issues. It's thumb rule, always load the JS after the DOM has loaded.

**Comment by ArpitKotecha** [comment]
> @ArpitKotecha, **add the script at end!**
> 
> Running `javascript` first may potentially cause performance issues. It's thumb rule, always load the JS after the DOM has loaded.

That's right but the analytics script is loaded asynchronously thus the browser makes a concurrent request for other assets. Also, Google recommends to put it right after head section so that it can track even when the user doesn't let the page finish loading. 
But if needed I can put it at the end of head or body tag.

**Comment by ArpitKotecha** [comment]
> That's awesome @ArpitKotecha !
> Once I'll get the credentials from @ShreyaAnand, I'll update `Google Analytics ID` and merge this.
> 
> Can you update the same for [vinnovateit.com](https://vinnovateit.com/) too!

Yes

**Comment by vinitshahdeo** [comment]
> > @ArpitKotecha, **add the script at end!**
> > Running `javascript` first may potentially cause performance issues. It's thumb rule, always load the JS after the DOM has loaded.
> 
> That's right but the analytics script is loaded asynchronously thus the browser makes a concurrent request for other assets. Also, Google recommends to put it right after head section so that it can track even when the user doesn't let the page finish loading.
> But if needed I can put it at the end of head or body tag.

I agree now **Google** has provided async support so that we get **accurate number of clicks** to our page. This might not cause any kind of performance issues util we don't have much client side rendering of components.

But the point here is we want users to spend some time on our website. Just by clicking and leaving the website even before `<body></body>` has loaded doesn't add any value to our use case. We want to track visitors who're interested in our own Hack - [VinHack](https://vinhack.vinnovateit.com/)!

This is my concern!

File: index.html [hunk 1]
```diff
@@ -2,6 +2,17 @@
 <html>
 
 <head>
+
+	<!-- Google analytics tag -->
+	<script async src="https://www.googletagmanager.com/gtag/js?id=UA-136271091-1"></script>
```
Comment (vinitshahdeo) [inline index.html]:
```suggestion
	<script async src="https://www.googletagmanager.com/gtag/js?id=UA-XXXXXXX"></script>
```

File: index.html [hunk 2]
```diff
@@ -2,6 +2,17 @@
...
+	<script>
+		window.dataLayer = window.dataLayer || [];
+		function gtag(){dataLayer.push(arguments);}
+		gtag('js', new Date());
+
+		gtag('config', 'UA-136271091-1');
```
Comment (vinitshahdeo) [inline index.html]:
```suggestion
		gtag('config', 'UA-XXXXXXX');
```

**Comment by ArpitKotecha** [comment]
> > > @ArpitKotecha, **add the script at end!**
> > > Running `javascript` first may potentially cause performance issues. It's thumb rule, always load the JS after the DOM has loaded.
> > 
> > 
> > That's right but the analytics script is loaded asynchronously thus the browser makes a concurrent request for other assets. Also, Google recommends to put it right after head section so that it can track even when the user doesn't let the page finish loading.
> > But if needed I can put it at the end of head or body tag.
> 
> I agree now **Google** has provided async support so that we get **accurate number of clicks** to our page. This might not cause any kind of performance issues util we don't have much client side rendering of components.
> 
> But the point here is we want users to spend some time on our website. Just by clicking and leaving the website even before `<body></body>` has loaded doesn't add any value to our use case. We want to track visitors who're interested in our own Hack - [VinHack](https://vinhack.vinnovateit.com/)!
> 
> This is my concern!

Well, in this case as I've made the requested changes. Although if we put Google Analytics script right after head tag we can know how many users are leaving because of website load time I guess we can use that in [Vinnovate](www.vinnovateit.com) website.

**Comment by ArpitKotecha** [comment]
Please ask @ShreyaAnand didi for google analytics credentials for [vinhack ](http://vinhack.vinnovateit.com)and [vinnovative ](http://vinnovateit.com)website

**Comment by vinitshahdeo** [comment]
> Please ask @ShreyaAnand didi for google analytics credentials for [vinhack ](http://vinhack.vinnovateit.com)and [vinnovative ](http://vinnovateit.com)website

Sure, I'll do!

File: index.html [hunk 3]
```diff
@@ -277,6 +278,20 @@ <h3 align="center">Fine Print Stationary</h3>
...
+	<script async src="https://www.googletagmanager.com/gtag/js?id=UA-XXXXXXXXX-X"></script>
+	<script>
+		window.dataLayer = window.dataLayer || [];
+
+		function gtag() {
+			dataLayer.push(arguments);
+		}
+		gtag('js', new Date());
+
+		gtag('config', 'UA-XXXXXXXXX-X');
```
Comment (vinitshahdeo) [inline index.html]: gtag('config', 'UA-136415898-1');

Use this `tracking ID`.

**Comment by ShreyaAnand** [comment]
Sent to @vinitshahdeo .

**Comment by vinitshahdeo** [comment]
`UA-136415898-1`

**Use this tracking ID!**

@ArpitKotecha 💯

**Comment by vinitshahdeo** [comment]
Can you resolve the conflicts? 

**Pull the latest `master` branch** and make the changes!

@ArpitKotecha

**Review by vinitshahdeo — COMMENTED** [review]
**Please resolve the conflicts!**

**Comment by ArpitKotecha** [comment]
> **Please resolve the conflicts!**

Sure. I'll resolve them.

**Comment by ArpitKotecha** [comment]
> > **Please resolve the conflicts!**
> 
> Sure. I'll resolve them.

Conflicts resolved @vinitshahdeo

**Review by vinitshahdeo — APPROVED** [review]
I'm merging it!
Good work @ArpitKotecha !

**Pull request merged**